libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/MapboxNavigationViewApi.kt

```diff
@@ -0,0 +1,105 @@
+package com.mapbox.navigation.dropin.internal
+
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.dropin.NavigationViewApi
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.destination.Destination
+import com.mapbox.navigation.ui.app.internal.destination.DestinationAction
+import com.mapbox.navigation.ui.app.internal.extension.dispatch
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterAction
+import com.mapbox.navigation.ui.app.internal.endNavigation as endNavigationAction
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class MapboxNavigationViewApi(
+    private val store: Store
+) : NavigationViewApi() {
+
+    override fun fetchRoutes(options: RouteOptions) {
+        store.dispatch(RoutePreviewAction.FetchOptions(options))
+    }
+
+    override fun fetchRoutes(points: List<Point>) {
+        store.dispatch(RoutePreviewAction.FetchPoints(points))
+    }
+
+    override fun setPreviewRoutes(routes: List<NavigationRoute>) {
+        store.dispatch(RoutePreviewAction.Ready(routes))
+    }
+
+    override fun setRoutes(routes: List<NavigationRoute>) {
+        store.dispatch(RoutesAction.SetRoutes(routes))
+    }
+
+    override fun setDestination(point: Point) {
+        store.dispatch(DestinationAction.SetDestination(Destination(point)))
+    }
+
+    override fun enableTripSession() {
+        store.dispatch(TripSessionStarterAction.EnableTripSession)
+    }
+
+    override fun enableReplaySession() {
+        store.dispatch(TripSessionStarterAction.EnableReplayTripSession)
+    }
```

Comment:
This function should never need to be called by the developer in production, whether and when the trip session is started should be controlled by the use-cases and the state machine in the Drop-In UI itself.

I understand that these are exposed only for debug purposes anyway, so I'd expect something more along the lines of:
```
fun routeReplayEnabled(boolean)
```
or
```
fun debugModeEnabled(boolean, flags)
```
where one of the flags is `ROUTE_REPLAY`.

Then, whether a regular trip session should be started, or the replay one, or switched between them, and all the other integration details would be handled by the Drop-In UI.

Reply:
Ok. Which method would you like me to implement?

Reply:
Replay state is tied to the trip session state at the moment..

so `routeReplayEnabled(false)` would be ambiguous (is the trip session enabled or disabled?).

If we assume routeReplayEnabled(false) keeps the trip session enabled it would be the same as `enableTripSession`

Reply:
My concern is why would a developer interacting with Drop-In UI need to know if the trip session is enabled or disabled, this should be controlled internally by the Drop-In UI, shouldn't it?
```
fun debugModeEnabled(boolean, flags)
```
proposal assumes that Drop-In starts replaying when appropriate (when we're in active navigation, and maybe simulate the first point in preview), developer only sets whether they want that or not.

Reply:
Our api is still driven by how things work to some extent. We have to pick layers, of where to put the abstraction conversions. Are we saying that the `DropInUi_api` layer should be unaware of trip sessions? 

If we change the abstraction of what this function actually does it requires new logic.

If we want to change `enableReplaySession ` to include a `routeReplayEnabled(false)` action, there should be a [TripSessionStarterAction](https://github.com/mapbox/mapbox-navigation-android/blob/da3115462b7e081ea25676e495a6f92adb785c69/libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/tripsession/TripSessionStarterAction.kt#L8-L24) that includes something like a `DisableRouteReplay` action with a definition for what is actually going to happen. I'm just calling out that it is not implemented.

Reply:
But now it is :)

``` kotlin
    override fun routeReplayEnabled(enabled: Boolean) {
        if (enabled) {
            store.dispatch(TripSessionStarterAction.EnableReplayTripSession)
        } else {
            store.dispatch(TripSessionStarterAction.EnableTripSession)
        }
    }
```

Reply:
But I also, don't find it a big deal so will let other opinions weigh in to decide

Reply:
This is another example of logic existing in the API Layer. If this is what we want, I also think the `TripSessionStarterAction` should eventually be updated to have a `RouteReplayEnabled(boolean)` action. That way the logic layer can have tests